modules/photons_control/planner/plans.py

```diff
@@ -816,6 +821,111 @@ async def info(self):
             return self.firmware
 
 
+@a_plan("hev_status")
+class HevStatusPlan(Plan):
+    """
+    Returns the current and previous HEV cycle run info for a device. The duration,
+    remaining and power_off fields are only included if there is an HEV cycle currently
+    running.
+
+        {
+            current: {
+                active: <boolean>,
+                duration: <total time in seconds>
+                remaining: <time left in seconds>
+                power_off: <boolean>
+            },
+            last: {
+                result: <result of last cycle>
+            }
+       }
+
+    * current:
+        * active: (boolean) true if the device is currently in a cycle
+        * duration: the total time in seconds for the current cycle
+        * remaining: the remaining time in seconds for the current cycle
+        * power_off: (boolean) true if the device power itself off when the cycle completes
+    * last:
+        * result: either SUCCESS, BUSY or INTERRUPTED BY <source>
+
+    """
+
+    default_refresh = 1
+
+    @property
+    def dependant_info(kls):
+        return {"c": CapabilityPlan()}
+
+    class Instance(Plan.Instance):
+        @property
+        def is_hev(self):
+            return self.deps["c"]["cap"].has_hev
+
+        @property
+        def messages(self):
+            if self.is_hev:
+                return [LightMessages.GetHevCycle(), LightMessages.GetLastHevCycleResult()]
+            return Skip
+
+        def process(self, pkt):
+            if pkt | LightMessages.StateHevCycle:
+                if pkt.remaining_s > 0:
+                    self.hev_current = {
+                        "active": True,
+                        "duration": pkt.duration_s,
+                        "remaining": pkt.remaining_s,
+                        "power_off": False if pkt.last_power == 1 else True,
+                    }
+                else:
+                    self.hev_current = {"active": False}
+
+            elif pkt | LightMessages.StateLastHevCycleResult:
+                self.hev_last = {"result": LightLastHevCycleResult(pkt.result).name}
```

Comment:
```suggestion
                self.hev_last = {"result": pkt.result}
```

I'm pretty sure that is guaranteed to be the enum when you access it and it's probably better to return it as the enum rather than the name.

Reply:
Ok